main.js

```diff
@@ -13,6 +13,275 @@ toggle.addEventListener('click', () => {
+    div2.appendChild(h3);
+    div2.appendChild(h3m);
+    div2.appendChild(ul);
+    div2.appendChild(btn);
+    card.appendChild(div2);
+    return card;
+  });
+  return card;
+}
+
+const cdContainer = createDiv(cardsData[0].class2);
+const cards = createCard(cardsData[1].divCls);
+cards.map((card) => cdContainer.appendChild(card));
+work.appendChild(cdContainer);
+
+const modal = document.querySelector('.popup');
+
+const arr = popupData[0].classes;
+const [demo, src] = popupData[0].btns;
+const [gitpages, github] = popupData[0].btnLinks;
+
+const [
+  close,
+  imgMob,
+  imgDesk,
+  popTtl,
+  seeLive,
+  seeLink,
+  popLinks,
+  link,
+  popPara,
+] = arr;
+const popDiv = createDiv2(popupData[0].id);
+const icon = document.createElement('i');
+icon.className = close;
+popDiv.appendChild(icon);
+const popImg = createDiv(popupData[0].divCls[0]);
+const img1 = createImg(popupData[0].images[0]);
+img1.className = imgMob;
+popImg.appendChild(img1);
+
+const img2 = createImg(popupData[0].images[1]);
+img2.className = imgDesk;
+popImg.appendChild(img2);
+popDiv.appendChild(popImg);
+modal.appendChild(popDiv);
+
+const poptext = createDiv(popupData[0].divCls[1]);
+const popHead = createDiv(popupData[0].divCls[2]);
+const h2pop = createH2(popupData[0].h2);
+h2pop.className = popTtl;
+popHead.appendChild(h2pop);
+
+const topBtnDiv = createDiv(popupData[0].divCls[3]);
+const btn1 = createBtn(popupData[0].btns[0]);
+btn1.className = seeLive;
+topBtnDiv.appendChild(btn1);
+
+const btn2 = createBtn(popupData[0].btns[1]);
+btn2.className = seeLink;
+topBtnDiv.appendChild(btn2);
+popHead.appendChild(topBtnDiv);
+poptext.appendChild(popHead);
+
+const ul = createUl(popLinks);
+const lists = createLists(popupData[0].lists);
+
+lists.map((item) => {
+  const elem = item;
+  elem.className = link;
+
+  return item.innerHTML === 'Bootstrap' || item.innerHTML === 'Terminal' || item.innerHTML === 'Codepen'
+    ? elem
+    : null;
+});
+
+lists.map((list) => ul.appendChild(list));
+poptext.appendChild(ul);
+
+const txt = createPara(popupData[0].para);
+txt.className = popPara;
+poptext.appendChild(txt);
+
+const bottomBtnDiv = createDiv(popupData[0].divCls[4]);
+const btn3 = createBtn(popupData[0].btns[2]);
+btn3.className = seeLive;
+btn3.innerHTML = demo;
+bottomBtnDiv.appendChild(btn3);
+
+const btn4 = createBtn(popupData[0].btns[3]);
+btn4.className = seeLink;
+btn4.innerHTML = src;
+bottomBtnDiv.appendChild(btn4);
+poptext.appendChild(bottomBtnDiv);
+popDiv.appendChild(poptext);
+modal.appendChild(popDiv);
+
+const cardBtn = document.querySelectorAll('.cardBtn');
+const body = document.querySelector('#portfolio');
+const hero = document.querySelector('.hero');
+const popup = document.querySelector('.popup');
+const collapse = document.querySelector('.fa-x');
+
+cardBtn.forEach((item) => {
+  item.addEventListener('click', () => {
+    popup.classList.add('open');
+    body.classList.add('blur');
+    hero.classList.add('blur');
+  });
+});
```

Comment:
- [ ] Your popup menu currently is not dynamically displaying the correct project clicked. I will ask that you will implement this. When a user clicks on a `see project` button, that particular project information is displayed in the popup dynamically. eg. 

```javascript
// Grab all button
const allSeeProjectBtn = document.querySelectorAll(".trans-btn");
// Attach an onClick event on each button like below:
//Before you have attached the project Id to each buttons or any other means to get the id of the project clicked
allSeeProjectBtn.forEach((button) => {
  button.addEventListener("click", (e) => {
    const projectId = e.target.id;    
    // find the project in the projects array usin the id.
    const currentProject = projects.find((project) => project.id === projectId);
    // Make the pop up appear
    displayPopup(currentProject); //You will need to create a function that handles displaying popup.
    // grab the close button for this popup using DOM manipulation connections such as node.previousElementSibling, etc  
  });
});

```